Code cleaning

delhomer opened the pull request:
Some cleaning operations to prepare the `0.4` release:
- version updatting in `setup.py`
- some useless modules deletion, some module renaming
- code reading (**in processing**)
- update `README.md`

Fixes the issue #63 .

garaud commented on setup.py:
```diff
@@ -6,7 +6,7 @@
             break
     else:
-        version = '0.3.2'
+        version = '0.4'
```
maybe write also the version number in the `__init__.py` file. What to you think?

delhomer replied:
Yes you're write, it seems to be a good practice. Should I modify the `setup.py` too, or should we keep the version in both scripts?

garaud replied:
we can keep the version in both scripts. I'll look for another open source projects how they deal with that.

delhomer committed:
clean: remove old legacy modules

delhomer committed:
cleaning: rename new modules in a proper way

delhomer committed:
cleaning: update the module imports with new filenames

delhomer committed:
training: fix checkpoint recovery in train.py

delhomer committed:
cleaning: define the generator seed at the beginning of the train.py module

delhomer committed:
cleaning: some code cleaning operation

Update docstrings and comments; remove print statements; remove
useless legacy functions in `utils.py`.

delhomer committed:
training: fix dropout argument

The dropout rate was wrongly implemented, as putting it as a command
argument did not change anything: the corresponding value was not
taken into account, and a 75% dropout rate was considered. This commit
fixes the issue, by setting dropout as an attribute of class
ConvolutionalNeuralNetwork and its derivatives.

delhomer committed:
versioning: update the program version to 0.4

delhomer committed:
cleaning: modify some argument default values

delhomer committed:
cleaning: update README.md for 0.4 release

garaud commented on README.md:
```diff
@@ -33,170 +28,167 @@ The picture below shows an example of image generated in this way:
++ numpy
++ pandas
++ PIL
++ tensorflow
```
maybe mention the setup.py file with the dependencies and the requitements dev file if needed.

garaud reviewed (changes requested):
minor comments

delhomer committed:
versioning: add project version in __init__.py

delhomer committed:
cleaning: add a short comment about dependencies in README.md

delhomer committed:
cleaning: add some metadata in setup.py

delhomer commented:
Some new commits fits your remarks. If everything is OK, we can merge, and release the new code version.
